src/main/java/ru/mts/Animals/AbstractAnimal.java

```diff
@@ -34,10 +58,86 @@ public String getName() {
     }
 
     @Override
-    public BigDecimal getCost() {return cost;}
+    public BigDecimal getCost() {
+        return cost;
+    }
 
     @Override
     public String getCharacter() {
         return character;
     }
+
+    @Override
+    public LocalDate getBirthDate() {
+        return birthDate;
+    }
+
+    /**
+     * <b>getFormatDate</b>
+     *
+     * @param format формат строки
+     * @return дату в формате format
+     */
+    public String getFormatDate(String format) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern(format);
+        return formatter.format(birthDate);
+    }
+
+    /**
+     * <b>equals</b>
+     *
+     * @param obj формат строки
+     * @return true или false в зависимости от равенства объектов
+     */
+    @Override
+    public boolean equals(Object obj) { // будут равны если равны имена, даты рождения и порода
+
+        if (this == obj) return true;
+
```

Comment:
```suggestion
        if (obj == null || getClass() != obj.getClass()) return false;
```
у тебя приведение типа проходит без проверки